File: eventbus-java/src/main/java/com/hluhovskyi/interviewkit/EventBus.java
package com.hluhovskyi.interviewkit;

@interface Subscribe {
}

interface EventBus {

    void register(Object subscriber);

    void unregister(Object subscriber);

    void post(Object event);

    static EventBus getInstance() {
        throw new InstantiationError("To be implemented");
    }
}

File: eventbus-kotlin/src/main/java/com/hluhovskyi/interviewkit/EventBus.kt
package com.hluhovskyi.interviewkit

annotation class Subscribe

interface EventBus {

    fun register(subscriber: Any)

    fun unregister(subscriber: Any)

    fun post(event: Any)

    companion object {

        fun getInstance(): EventBus = throw InstantiationError("To be implemented")
    }
}
